app/models/organization.rb

```diff
@@ -9,7 +9,8 @@ class Organization < ApplicationRecord
 
   belongs_to :business_plan
   belongs_to :contact_person, class_name: 'User', optional: true
-  has_many :users, class_name: 'User', dependent: :destroy
+  has_many :users_organizations, dependent: :destroy
+  has_many :users, through: :users_organizations, dependent: :restrict_with_exception
```

Comment:
can't we just leave the dependent param out. To my understanding the users is just kind of a shortcut going through the users_organization. If the users_organizations is destroyed (which is correct) then there is nothing to be done at the users at all.

Reply:
We can. I thought that rubocop would complain and I think that raising an exception is a valid thing to do in this case, but the default is nil, which also makes sense. The thing is there is should be no UI to delete a user as this  leads to an exception if there are any requests by that user.

```
ActiveRecord::InvalidForeignKey: PG::ForeignKeyViolation: ERROR:  update or delete on table "users" violates foreign key constraint "fk_rails_8ead8b1e6b" on table "requests"
DETAIL:  Key (id)=(5) is still referenced from table "requests".
```

There is also no UI to delete an organization.

See 8e2b1fb0